libraries/SPI/SPI.cpp

```diff
@@ -74,13 +75,17 @@ void SPIClass::setHwCs(bool use) {
 }
 
 void SPIClass::beginTransaction(SPISettings settings) {
-    while(SPI1CMD & SPIBUSY) {}
+    while(inTransaction || SPI1CMD & SPIBUSY) {
+        yield();
```

Comment:
the SPI itself is interrupt safe, but you add a ```yield``` and that is not.
all function that can trigger a task switch like ```yield``` and ```delay``` are designed to be only on the arduino task (setup or loop), if you call them in a interrupt the system will crash.
https://github.com/esp8266/Arduino/blob/master/doc/reference.md#timing-and-delays